src/VisualStudio/TestUtilities/VisualStudioInstance.cs

```diff
@@ -122,6 +132,9 @@ internal Task<Window> LocateDteWindowAsync(string windowTitle)
         internal Task WaitForDteCommandAvailabilityAsync(string command)
             => IntegrationHelper.WaitForResultAsync(() => IntegrationHelper.RetryDteCall(() => Dte.Commands.Item(command).IsAvailable), expectedResult: true);
 
+        public void WaitForApplicationIdle()
```

Comment:
I do not think there should be a wait API like this at this layer. All waits should live in code that's running in VS, as otherwise it'll be unclear where the waits need to happen.